### ISSUE TITLE
Prepare MPTelemetry-1.0 for GA

### ISSUE THREAD

**Issue by Yas-Franco** [OpenLiberty/open-liberty#22944]
The implementation passes the [RC-5 TCK](https://github.com/eclipse/microprofile-telemetry/pull/61). Before GA, we need to: 

- [x] Move from OTel 1.18 to 1.19
- [x] Add FAT tests for API calls not covered by TCK: 
  - [x] ~Test GlobalOpenTelemetry.get()~ removed from spec
  - [x]  Test Span.current()
  - [x]  Test Baggage.current()
- [x] Fix the error thrown by Java 2 Security causing [defect 292435](https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=292435).
- [x] #23206
- [x] Check use of `Instrumenter` from [opentelemetry-java-instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation) agent in our integration with restfulWS and rest client.
  - I think these are meant to be API for use by components of the instrumentation agent, but we're using them outside the agent
  - We should check why we're using this over the regular [opentelemetry API](https://opentelemetry.io/docs/instrumentation/java/manual/)
- [x] #23080 
- [x] #23177 
- [x] #23176
- [x] #23175
- [x] #23447
- [x] #23448 
- [x] #23450 
- [x] #23451 
- [x] #23452 
- [x] #23453 
- [x] #23456
- [x] #23461 
- [x] #19752
- [x] #23521
- [x] #23670
- [x] #23736 
- [x] #23923 
- [x] #24139
- [x] #24226
- [x] #24176 
- [x] #24273
- [x] #24044 
- [x] #23521
- [x] #24372 

Before release but after approvals given:
- [x] #24045 

Not essential for release:
- [x] #23455 
- [x] #23457 
- [x] #23458 
- [x] #23489

**Comment by Azquelt** [comment]
Re: use of `Instrumenter`

We used `Instrumenter` to take advantage of the deduplication of spans in the Agent, allowing us to have our integration enabled while the agent is running and have it automatically deduplicate spans.

However, as documented in #23640, this doesn't quite work as we'd hoped:
- spans added for `@WithSpan` don't get deduplicated
  - the agent code does not implement the necessary interface to enable deduplication here
  - additionally, I'm not sure it's possible to tell the difference between two spans created for one method call which should be deduplicated and two spans created for a recursive method call which shouldn't be deduplicated
- the agent allows the user to disable span creation at specific points using config. However, if they turn off spans for rest requests, they still get our rest integration, which is confusing.

For these reasons, we decided to disable our integration for REST and `@WithSpan` when the agent is enabled.

This means we no longer _need_ to use the `Instrumentor` API, though it does provide some helpful abstraction of the exact naming of attributes. Unless there's a pressing reason, I don't think we need to do the work required to switch away from the `Instrumentor` API.
